WordPress/gutenberg
# Legacy Widget: Improve backwards compatibility

## Description
Adds back two things that I removed in https://github.com/WordPress/gutenberg/pull/29960 but shouldn't have removed 🙂

- Fire 'widget-added' and 'widget-updated' events when the widget form is initialized or when it changes. This is what most widgets' scripts use to initialize and/or attach event listeners.
- Include hidden inputs that most widgets expect to exist in the DOM.

## How has this been tested?
1. Add this somewhere (e.g. to the bottom of `gutenberg.php`) so that you can test the Legacy Widget block with all core widgets.
    ```php
   add_filter(
          'widget_types_to_hide_from_legacy_widget_block',
          function() {
                  return array();
          }
   );
   ```
2. Go to Appearance → Widgets.
3. Insert a Legacy Widget block.
4. Select the Text widget. Play with it. It should work.

## Screenshots
Before:
<img width="680" alt="Screen Shot 2021-04-12 at 12 27 53" src="https://user-images.githubusercontent.com/612155/114332949-8ab76900-9b8a-11eb-89bc-06fe73e51d1c.png">

After:
<img width="684" alt="Screen Shot 2021-04-12 at 12 27 30" src="https://user-images.githubusercontent.com/612155/114332944-855a1e80-9b8a-11eb-8c80-8b5433678b7a.png">

## Types of changes
Bug fix

## Checklist:
